roqoqo-qasm/src/backend.rs

```diff
@@ -40,6 +40,8 @@ pub struct Backend {
     /// When translating to QASM which uses explicitely declared qubit registers a name for the qubit
     /// register needs to be chosen.
     qubit_register_name: String,
+    /// Which version of OpenQASM (2.0 or 3.0) to use
+    qasm_version: String,
```

Comment:
We could also think about using an enum of supported qasm versions here. That would have the advantage, that we can sanitize the input early and we have all the possible supported version encoded in the type system.